src/Driver/AbstractDriver.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+
+namespace Drift\DBAL\Driver;
+
+
+use Doctrine\DBAL\Query\QueryBuilder;
+use React\Promise\PromiseInterface;
+
+abstract class AbstractDriver
+{
+    abstract public function query(string $sql, array $parameters): PromiseInterface;
```

Comment:
This method is already defined for implementation in the Driver interface. Because this class is abstract, you don't have to implement it.

Reply:
If all drivers have to extend this `AbstractDriver`, then the interface can be implemented directly in this abstract class and not in the specific classes (the contract is implicit when you extend)